drivers/wifi/nrf700x/zephyr/src/qspi/src/qspi_if.c

```diff
@@ -612,7 +660,22 @@ static int qspi_nrfx_configure(const struct device *dev)
 
 	qspi_fill_init_struct(&QSPIconfig);
 
+#if defined(CONFIG_SOC_SERIES_NRF53X)
+	/* When the QSPI peripheral is activated, during the nrfx_qspi driver
+	 * initialization, it reads the status of the connected flash chip.
+	 * Make sure this transaction is performed with a valid base clock
+	 * divider.
+	 */
+	nrf_clock_hfclk192m_div_set(NRF_CLOCK, BASE_CLOCK_DIV);
+#endif
+
 	nrfx_err_t res = _nrfx_qspi_init(&QSPIconfig, qspi_handler, dev_data);
+
+#if defined(CONFIG_SOC_SERIES_NRF53X)
+	/* Restore the default /4 divider after the QSPI initialization. */
+	nrf_clock_hfclk192m_div_set(NRF_CLOCK, NRF_CLOCK_HFCLK_DIV_4);
```

Comment:
In case `CONFIG_NRF700X_QSPI_LOW_POWER` is defined we put the QSPI in low power mode after its use in `qspi_device_uninit`, so, we should also do this there.

Reply:
There is done in the qspi_unlock(). Will send a mail detailing my studies.